README.md

```diff
@@ -1064,545 +983,270 @@ helpLinks: {
-                    "errorSummary": "Custom field level error",
-                    "reason": "registration.error.address",
-                    "resource": "User",
-                    "property": "address", //should match field name
-                    "arguments": []
-                }
-            ]
-        };
-        onFailure(error);
-     }
-    ```
-
-### IdP Discovery
+  }
+},
+```
 
-IdP Discovery enables you to route users to different 3rd Party IdPs that are connected to your Okta Org. Users can federate back into the primary org after authenticating at the IdP.
+#### parseSchema
 
-To use IdP Discovery in your application, configure an [identity provider routing rule](https://help.okta.com/en/prod/Content/Topics/Security/configure-routing-rules.htm) in the Okta admin panel.
-Then, in the widget configuration, set `features.idpDiscovery` to `true`.
+Callback used to change the JSON schema that comes back from the Okta API.
 
 ```javascript
-var signIn = new OktaSignIn({
-  baseUrl: 'https://{yourOktaDomain}',
-  ... ...
-  features: {
-    idpDiscovery: true
-  }
-});
+parseSchema: (schema, onSuccess) => {
+  // This example will add an additional field to the registration form.
+  schema.push(
+    {
+      'name': 'userProfile.address',
+      'type': 'text',
+      'placeholder': 'Enter your street address',
+      'maxLength': 255,
+      'label-top': true,
+      'label': 'Street Address',
+      'required': true,
+    }
+  );
+  onSuccess(schema);
+}
 ```
 
-**Note:** IdP Discovery is supported for Okta-hosted Widget setup only. IdP Discovery for self-hosted Widget is not officialy supported by v1 API and requires an extra step to complete OIDC flow. This step involves calling [`session.exists`](https://github.com/okta/okta-auth-js#sessionexists) and [`token.getWithoutPrompt`](https://github.com/okta/okta-auth-js#tokengetwithoutpromptoptions) methods which rely on third party cookies to be available. The below snippet applies to a SPA app hosting Sign-In Widget:
+#### preSubmit
 
-```javascript
-var signIn = new OktaSignIn({
-  baseUrl: 'https://{yourOktaDomain}',
-  clientId: '{appClientId}',
-  ... ...
-  features: {
-    idpDiscovery: true
-  },
-  idpDiscovery: {
-    requestContext: window.location.href
-  }
-});
-
-... ...
+Callback used primarily to modify the request parameters sent to the Okta API.
 
-// after successful authentication with IdP
-signIn.authClient.session.exists().then(function (sessionExists) {
-  if (sessionExists) {
-    signIn.authClient.token.getWithoutPrompt().then(function (response) {
-      signIn.authClient.tokenManager.setTokens(response.tokens);
-    });
+ ```javascript
+preSubmit: (postData, onSuccess) => {
+  // This example will append the domain name to the email address if the user forgets to add it during registration.
+  if (postData.userProfile.email.includes('@acme.com')) {
+    postData.userProfile.email;
+  } else {
+    postData.userProfile.email + '@acme.com';
```

Comment:
This is incorrect, and my bad since I wrote this, 
```suggestion
  if (!postData.userProfile.email.includes('@acme.com')) {
    postData.userProfile.email += '@acme.com';
  }
```